### PR TITLE
update union_relations to use core string literal macro

### LINKED ISSUE #664: `union_relations` uses deprecated `dbt_utils.string_literal()`
### Describe the bug

dbt_utils.union_relations contains a reference to the `dbt_utils.string_literal()` macro, rather than the dbt-core implementation, causing the deprecation warning to be triggered

### Steps to reproduce

Install latest dbt_utils, use in a model:

```sql
{{
  dbt_utils.union_relations(
    [
        ref('model_1'),
        ref('model_2')
    ]
  )  
}}
```

### Screenshots and log output

image

### Expected results

- No warnings to appear for models using this macro

### System information
**The contents of your `packages.yml` file:**
```
packages:
  - local: "../../../dbt-packages/dbt-utils"
```
(main branch checked out over yonder ^)

**Which database are you using dbt with?**
- [x] postgres
- [ ] redshift
- [ ] bigquery
- [x] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
Core:
  - installed: 1.2.1
  - latest:    1.2.1 - Up to date!

Plugins:
  - snowflake: 1.2.0 - Up to date!
  - postgres:  1.2.1 - Up to date!
```

### Are you interested in contributing the fix?
heck yes

### PR DESCRIPTION
**Pull request by dave-connors-3**
resolves #664 

This is a:
- [ ] documentation update
- [x] bug fix with no breaking changes
- [ ] new functionality
- [ ] a breaking change

All pull requests from community contributors should target the `main` branch (default).

## Description & motivation

This updates the `string_literal` cross db macro in `union_relations` to use the core version rather than the dbt_utils version to avoid deprecation warnings. 

### Before 

image

### After

image

## Checklist
- [x] This code is associated with an Issue which has been triaged and [accepted for development](https://docs.getdbt.com/docs/contributing/oss-expectations#pull-requests). 
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [x] Postgres
    - [ ] Redshift
    - [ ] Snowflake
- [x] I followed guidelines to ensure that my changes will work on "non-core" adapters by:
    - [ ] dispatching any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/dbt-labs/dbt-utils/blob/main/macros/sql/star.sql))
    - [ ] using the `limit_zero()` macro in place of the literal string: `limit 0`
    - [ ] using `dbt.type_*` macros instead of explicit datatypes (e.g. `dbt.type_timestamp()` instead of `TIMESTAMP`
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Comment by dbeatty10** [comment]
@dave-connors-3 I'm _guessing_ that CI is failing because of [this](https://github.com/dbt-labs/dbt-bigquery/pull/303#issuecomment-1244346023), but I don't know for certain.

**Comment by dave-connors-3** [comment]
@dbeatty10 have we noticed anything similar for other PRs?

**Comment by dbeatty10** [comment]
The `main` branch of dbt-bigquery is broken right now:
image

Since the CI for [dbt-utils uses the `main` branch of dbt-bigquery](https://github.com/dbt-labs/dbt-utils/blob/53f8352a2d93643aeb4e2b1e3de8e49b7364d00d/dev-requirements.txt#L8), I'm guessing you'll need to wait for that to be fixed before re-running the tests for this PR.

**Review by joellabes — APPROVED** [review]
I approve the changes, but let's wait until BigQuery is passing to actually merge this in 👍

**Comment by dbeatty10** [comment]
[This](https://github.com/dbt-labs/dbt-bigquery/pull/305/files) was the PR that got things passing again.

Should be good-to-go now.

**Pull request merged**
